### PR TITLE
[guides] Add a guide on writing document titles and headings

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fixes ENG-13207

# How

Add a guide on how to write document titles and section headings that can be optimized for search.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: guides/Wrtiting scannable titles and headings.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+---
+
+Let's start with an example where a heading is unaligned to its document's content:
+
+[<img src="./assets/writing-guides/01-search-doc-example.png" width="540" />](./assets/writing-guides/01-search-doc-example.png)
+
+Based on the search term above, the correct page appears as the last item. The search term contains all keywords that are common with the way we talk about running an EAS Build on a local computer.
+
+Using the context of a keyword (commonly used term), it improves the search ranking of the document when the same search term is used:
```
Comment (keith-kurak) [inline guides/Wrtiting scannable titles and headings.md:15]:
```suggestion
Using keywords in the heading improves the search ranking of the document when the same search term is used:
```

File: guides/Wrtiting scannable titles and headings.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+## General tips
+
+When writing a document title or a section heading, do the following:
+
+- Be concise.
+- Describe the main purpose of the document or the section
+- For procedural documents (guides and tutorials), use a verb-based title/heading to describe the action that you expect the developer to perform. Using an action verb also informs the developer about the type of content they can expect to find once they land on the page and start scanning it.
+- Front-load keywords in headings to keep the important information or subject at the beginning.
+- Focus on what matters most to customers and choose words they would use themselves.
+- Use sentence case for a section or sub-section headings. For more details, see [formatting guidelines for headings](./Expo%20Documentation%20Writing%20Style%20Guide.md#headings)
```
Comment (keith-kurak) [inline guides/Wrtiting scannable titles and headings.md:30]:
```suggestion
- Use sentence case (i.e., capitalize the first word and proper nouns only) for section or sub-section headings. For more details, see [formatting guidelines for headings](./Expo%20Documentation%20Writing%20Style%20Guide.md#headings)
```

File: guides/Wrtiting scannable titles and headings.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+The improvement above also tells the developer about the primary purpose of the document on first glance.
+
+## General tips
+
+When writing a document title or a section heading, do the following:
+
+- Be concise.
+- Describe the main purpose of the document or the section
+- For procedural documents (guides and tutorials), use a verb-based title/heading to describe the action that you expect the developer to perform. Using an action verb also informs the developer about the type of content they can expect to find once they land on the page and start scanning it.
```
Comment (keith-kurak) [inline guides/Wrtiting scannable titles and headings.md:27]:
```suggestion
- For procedural documents (guides and tutorials), use a verb-based title/heading (e.g., "Run EAS Build") to describe the action that you expect the developer to perform. Using an action verb also informs the developer about the type of content they can expect to find once they land on the page and start scanning it.
```

File: guides/Wrtiting scannable titles and headings.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- Front-load keywords in headings to keep the important information or subject at the beginning.
+- Focus on what matters most to customers and choose words they would use themselves.
+- Use sentence case for a section or sub-section headings. For more details, see [formatting guidelines for headings](./Expo%20Documentation%20Writing%20Style%20Guide.md#headings)
+
+### Document titles
+
+When using multiple content types in the same document, such as a section or two to describe a procedure that is followed by a section that explains the concept, use a task-based title (action-based verbs) for that document.
```
Comment (keith-kurak) [inline guides/Wrtiting scannable titles and headings.md:34]: If task-based title is the same as the "verb-based" title/heading above, let's us the same words here.

**Review by keith-kurak — APPROVED** [review]
Added some ideas for clarifying what works, but looks good, let's get it out there!

File: guides/Wrtiting scannable titles and headings.md (hunk 1 above)
Comment (amandeepmittal) [reply guides/Wrtiting scannable titles and headings.md:15]: not exactly; I meant to say here that some sort of context that represents the keyword. A keyword in every case will not be an ideal fit. I'll try rephrase it.

File: guides/Wrtiting scannable titles and headings.md (hunk 2 above)
Comment (amandeepmittal) [reply guides/Wrtiting scannable titles and headings.md:30]: I think it's more than just proper nouns; it's also branding and product/service names. This is why I linked to writing style guide doc's section: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings

File: guides/Wrtiting scannable titles and headings.md (hunk 4 above)
Comment (amandeepmittal) [inline guides/Wrtiting scannable titles and headings.md:34]:
```suggestion
When using multiple content types in the same document, such as a section or two describing a procedure followed by a section explaining the concept, use a verb-based title for that document.
```

**Pull request merged**